ompi/mca/topo/base/topo_base_cart_create.c

```diff
@@ -43,13 +43,19 @@ static int mca_topo_base_cart_allocate (ompi_group_t *group, int ndims, const in
     /* Calculate the number of processes in this grid */
     for (int i = 0 ; i < ndims ; ++i) {
         if (dims[i] <= 0) {
+            if (NULL != cart) {
```

Comment:
Should we check for `cart != NULL` at the top of the function?

Reply:
That would be cleaner, but I wasn't sure if that was going to cause problems because the my_rank and num_procs parameters can be updated just above the current check for cart != NULL.

I was being careful not to change the behavior since I don't know if callers are making assumptions about these parameters that a change would break.

I can move the check if you think it's safe.

Reply:
Actually, this function is only called from mca_topo_base_cart_create and the passed num_procs and new_rank are local variables to that function, so this change is safe and I will make the change.

Reply:
Did you make that change? I don't see anything different...